### DISCUSSION TITLE
ECS Cluster ASG drain hook uses python 3.6 runtime which will not be supported by AWS soon.

### DISCUSSION THREAD

**Discussion by baraksalomon** [aws/aws-cdk#20194] [Q&A]
Hello,

The Lambda created as part of the drain hook for ECS ASG uses python 3.6 runtime:
https://github.com/aws/aws-cdk/blob/v1.154.0/packages/@aws-cdk/aws-ecs/lib/drain-hook/instance-drain-hook.ts
is there a plan to change it to a supported python runtime version?

Thanks,
       Barak.

**Comment by PiotrJander** [comment]
This appears that a newer version of CDK (I checked with 2.65) fixes this by using Python 3.9
